databox/api/src/Integration/Phrasea/Expose/ExposeClient.php

```diff
@@ -149,13 +151,86 @@ public function postAsset(IntegrationConfig $config, IntegrationToken $integrati
             ;
             $exposeAssetId = $pubAsset['id'];
 
-            $this->uploadClient->request('PUT', $pubAsset['uploadURL'], [
-                'headers' => [
-                    'Content-Type' => $source->getType(),
-                    'Content-Length' => filesize($fetchedFilePath),
-                ],
-                'body' => fopen($fetchedFilePath, 'r'),
-            ]);
+            $uploadsData = [
+                'filename' => $source->getOriginalName(),
+                'type' => $source->getType(),
+                'size' => (int)$source->getSize(),
+            ];
+
+            $resUploads = $this->create($config, $integrationToken)
+                ->request('POST', '/uploads', [
+                    'json' => $uploadsData,
+                ])
+                ->toArray()
+            ;
+
+            $mUploadId = $resUploads['id'];
+            $parts['Parts'] = [];
+            $isPartsComplete = false;
+           
+            // Upload the file in parts.
+            try {
+                $file = fopen($fetchedFilePath, 'r');
+                $partNumber = 1;
+                while (!feof($file)) {
+                    $resUploadPart = $this->create($config, $integrationToken)
+                        ->request('POST', '/uploads/'. $mUploadId .'/part', [
+                            'json' => ['part' => $partNumber],
+                        ])
+                        ->toArray()
+                    ;
+
+                    try {
+                        $headerPutPart = $this->uploadClient->request('PUT', $resUploadPart['url'], [
+                                'body' => fread($file, 10 * 1024 * 1024), // 10Mo
+                            ])->getHeaders()
+                        ;
+
+                    } catch (\Exception $e) {
+                        sleep (1); // retry after 1 second
+                        try {
+                            $headerPutPart = $this->uploadClient->request('PUT', $resUploadPart['url'], [
+                                'body' => fread($file, 10 * 1024 * 1024), // 10Mo
+                            ])->getHeaders()
+                        ;
+                        } catch (\Exception $e) {
```

Comment:
```suggestion
                        } catch (\Throwable $e) {
```
Please check others!